---
title: "01_McCraken"
author: "Sera Choi"
date: "12/07/2021"
output: html_document
---

```{r, message=FALSE, warning=FALSE}
# Load library
library(Seurat)
library(ggplot2)
library(dplyr)
library(here)
library("scales")
library(ggsci)
library(gtools)
library(future)

mypal <- pal_npg("nrc", alpha = 0.7)(10)
mypal2 <-pal_tron("legacy", alpha = 0.7)(7)
mypal3 <- pal_lancet("lanonc", alpha = 0.7)(9)
mypal4 <- pal_simpsons(palette = c("springfield"), alpha = 0.7)(16)
mypal5 <- pal_rickandmorty(palette = c("schwifty"), alpha = 0.7)(6)
mypal6 <- pal_futurama(palette = c("planetexpress"), alpha = 0.7)(5)
mypal7 <- pal_startrek(palette = c("uniform"), alpha = 0.7)(5)
mycoloursP<- c(mypal, mypal2, mypal3, mypal4, mypal5, mypal6, mypal7)
show_col(mycoloursP, labels =F)

# Set maximum size of the object 
options(future.globals.maxSize = 50000*1024^2)
```

# McCraken et al (GSE131736)

This part of the project compares single-cell transcriptomisc of ES-derived ECs vs Young adult ECs to understand the suitability of ES-derived ECs as model cells for brain ECs and to improve current differenciation protocol.  

```{r}
counts <- Read10X_h5("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/Craken_1.h5", use.names = TRUE, unique.features = TRUE)
seur <- CreateSeuratObject(counts = counts, assay = "RNA",min.cells = 3, min.features = 200)
```

## Quality control

Using thresholds Craken et al used, cells with 

* cells expressing < 250 genes 
* cells having UMI count > 10,000
* cells with > 15% mito genes 

were removed. 

```{r}
seur[["percent.mt"]] <- PercentageFeatureSet(seur, pattern = "^MT-")
seur[["orig.ident"]] <- "Craken_1"
VlnPlot(seur, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, pt.size =0)

# Apply QC threshold
seur <- subset(seur, subset = nCount_RNA > 250 & nFeature_RNA < 10000 & percent.mt < 15)

#Visualise
VlnPlot(seur, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, pt.size =0)
plot1 <- FeatureScatter(seur, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(seur, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1 + plot2
```

## Normalisation, 

```{r}
# Normalisation
seur <- NormalizeData(seur, normalization.method = "LogNormalize", scale.factor = 10000)

# Feature Selection
seur <- FindVariableFeatures(seur, selection.method = "vst", nfeatures = 2000)

# Scaling 
all.genes <- rownames(seur)
seur <- ScaleData(seur, features = all.genes)

# Linear dimensional reduction
seur <- RunPCA(seur, features = VariableFeatures(object = seur), verbose = FALSE)
ElbowPlot(seur, ndims=50)

# Visualisation
seur <- RunUMAP(seur, dims = 1:20, verbose = FALSE)
DimPlot(seur, reduction = "umap")

# CLustering 
seur <- FindNeighbors(seur, dims = 1:20, verbose = FALSE)
seur <- FindClusters(seur, resolution = 0.01, verbose = FALSE)
DimPlot(seur, reduction = "umap")

saveRDS(seur, "/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/Craken.RDS")
```

## Cell-type characterisation

Using gene markers Craken et al used:

* pluripotent (POUF51, SOX2, NANOG)
* mesodermal (HAND1, T, MIXL1)
* mesenchymal (ACTA2, ALCAM, PDGFRB)
* endothelial (PECAM1, KDR, CD34, CDH5)
* epithelial (EPCAM, CLDN4, RAB25, WNT6)
* nephrogenic (PAX2, PAX8, CRYM)


```{r, fig.width=10}
# endothelial 
FeaturePlot(seur, features = c('PECAM1', 'KDR', 'CD34', 'CDH5'))
# Mesenchymal
FeaturePlot(seur, features = c('ACTA2', 'ALCAM', 'PDGFRB'))
```

## Subeset ECs

2,896 ECs

```{r}
Idents(seur) <- "RNA_snn_res.0.01"
es <- subset(seur, idents = 0)
saveRDS(es, "/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/Craken_ec.RDS")
```

## Luise

The comparison of ES-derived ECs will be made with Luise's Young adult cells (30-65).

```{r}
luise <- readRDS("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/Luise_ec.RDS")
es <- readRDS("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/Craken_ec.RDS")

# Classify age group
mod_metadata <- luise@meta.data  %>% mutate(AgeGroup = case_when(Age < 65 ~ 'Young',
                                                                 Age > 65 ~ 'Old'))
                                    
luise[["AgeGroup"]] <- mod_metadata$AgeGroup

# Subset Young cells 
Idents(luise) <- "AgeGroup"
luise <- subset(luise, idents = "Young")
luise[["orig.ident"]] <- "Luise"
es[["orig.ident"]] <- "McCracken"
```

## Integrate Craken and Luise

```{r}
comb <- merge(x = luise, y = es, add.cell.ids = c("Endothelial", "ES_derived"))

# Set up Seurat object list, run SCTransform oneach object separately
ec.list <- SplitObject(comb, split.by = "orig.ident")
for (i in 1:length(ec.list)) {
    ec.list[[i]] <- SCTransform(ec.list[[i]], verbose = F)
}

# Select features for downstream integration
ec.features <- SelectIntegrationFeatures(object.list = ec.list, nfeatures = 2000)
# Ensures that all necessary Pearson residuals have been calculated 
ec.list <- PrepSCTIntegration(object.list = ec.list, anchor.features = ec.features, verbose = F)

# identify anchors and integrate datasets 
ec.anchors <- FindIntegrationAnchors(object.list = ec.list, normalization.method = "SCT", anchor.features = ec.features, verbose = F)
ec.integrated <- IntegrateData(anchorset = ec.anchors, normalization.method = "SCT", verbose = F)
```


## Clustering and visualisation 

2,896 ES-derived ECs and 416 mature ECs

```{r}
ec.integrated <- RunPCA(ec.integrated, verbose = FALSE)
ElbowPlot(ec.integrated, ndims = 50)

ec.integrated <- RunUMAP(ec.integrated, dims = 1:20, verbose=FALSE)
DimPlot(ec.integrated, reduction = "umap", group.by = "orig.ident", cols=mycoloursP[1:25])
table(ec.integrated@meta.data$orig.ident)
```

## Downsampling Craken

```{r}
reduced_es <- subset(es, downsample = 500)

```

## Integrate with downsampled Craken 

```{r}
comb <- merge(x = luise, y = reduced_es, add.cell.ids = c("Endothelial", "ES_derived"))

# Set up Seurat object list, run SCTransform oneach object separately
ec.list <- SplitObject(comb, split.by = "orig.ident")
for (i in 1:length(ec.list)) {
    ec.list[[i]] <- SCTransform(ec.list[[i]], verbose = F)
}

# Select features for downstream integration
ec.features <- SelectIntegrationFeatures(object.list = ec.list, nfeatures = 2000)
# Ensures that all necessary Pearson residuals have been calculated 
ec.list <- PrepSCTIntegration(object.list = ec.list, anchor.features = ec.features, verbose = F)

# identify anchors and integrate datasets 
ec.anchors <- FindIntegrationAnchors(object.list = ec.list, normalization.method = "SCT", anchor.features = ec.features, verbose = F)
ec.integrated <- IntegrateData(anchorset = ec.anchors, normalization.method = "SCT", verbose = F)

# Visualisation
ec.integrated <- RunPCA(ec.integrated, verbose = FALSE)
ElbowPlot(ec.integrated, ndims = 50)

ec.integrated <- RunUMAP(ec.integrated, dims = 1:20)
DimPlot(ec.integrated, reduction = "umap", group.by = "orig.ident", cols=mycoloursP[1:25])
table(ec.integrated@meta.data$orig.ident)
```

## QC of integrated data

```{r}
VlnPlot(ec.integrated, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
plot1 <- FeatureScatter(ec.integrated, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(ec.integrated, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1 + plot2
```

## Downstream analysis 

```{r}
# Normalise 
ec.integrated <- NormalizeData(ec.integrated, verbose = FALSE, assay = "RNA")

# PCA analysis
Idents(ec.integrated) <- "orig.ident"
ec.integrated <- RunPCA(ec.integrated, verbose=FALSE)
DimPlot(ec.integrated, reduction = "pca")
ec.integrated <- RunUMAP(ec.integrated, dims = 1:20, verbose=FALSE)
DimPlot(ec.integrated, reduction = "umap")

saveRDS(ec.integrated, "/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/ES_EC_integrated.RDS")
```




















