---
title: "07_age_Luise"
author: "Sera Choi"
date: "13/06/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, message=FALSE, warning=FALSE}
# Load library
library(Seurat)
library(ggplot2)
library(ggsci)
library(gtools)
library(dplyr)
library(here)
library("scales")
library(future)

mypal <- pal_npg("nrc", alpha = 0.7)(10)
mypal2 <-pal_tron("legacy", alpha = 0.7)(7)
mypal3 <- pal_lancet("lanonc", alpha = 0.7)(9)
mypal4 <- pal_simpsons(palette = c("springfield"), alpha = 0.7)(16)
mypal5 <- pal_rickandmorty(palette = c("schwifty"), alpha = 0.7)(6)
mypal6 <- pal_futurama(palette = c("planetexpress"), alpha = 0.7)(5)
mypal7 <- pal_startrek(palette = c("uniform"), alpha = 0.7)(5)
mycoloursP<- c(mypal, mypal2, mypal3, mypal4, mypal5, mypal6, mypal7)
show_col(mycoloursP, labels =F)

# Set maximum size of the object 
options(future.global.maxSize = 16000*1024^2)

# Load data
ec <- readRDS(here("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/Luise_ec.RDS")) 
```

# Differential gene expression with age

This section performs DE analysis on ECs of young and old brain tissues. 


## Enhanced genes in Old 

Overall, 7 genes are expressed higher in old than young that satisfy adjusted p-value < 0.05. Of these, only two genes satisfy the pct.1, pct.2 and adjusted p-value thresholds (ACER2, FGD6). These 7 genes are genes I would like to explore further and also test on other datasets. Although the difference between young and aged is significant, the difference in expression is not large. The largest log2FC is 0.6, which is 2^0.6 = 1.5. This may be because there are not many ECs in the dataset. So, here I chose to filter by pct1 and pct2 as log2FC is small. To choose markers to validate in lab, ACER2 and FGD6 would be used but also, PLAT and TNRC6B could also be used.  

**SLC7A5 (LAT1)**

* Amino acid transporter responsible for uptake of amino acids in BBB (Capillary EC)
* Involved in cancer and neurological disorders
* An important pharmacological target 
* As an essential amino acid transporter, has been reported to be enhanced in variety of cancers. It supplies amino acids to cancer cells 
* Pathogenic mutations cause microcephaly and seizures
* The Human SLC7A5 (LAT1): The Intriguing Histidine/Large Neutral Amino Acid Transporter and Its Relevance to Human Health(2018), The amino acid transporter SLC7A5 is required for efficient growth of KRAS-mutant colorectal cancer (2021), The amino acid transporter SLC7A5 confers a poor prognosis in the highly proliferative breast cancer subtypes and is a key therapeutic target in luminal B tumours (2018)  

**CDK17**

* CDK17 upregualted in Tau and Oligo (Quantitative phosphoproteomics uncovers dysregulated kinase networks in Alzheimerâ€™s disease, 2020)
* CDKs are responsible for controlled cell proliferation. 

**ACER2**

* Alkaline ceramidase 2 enhanced in capillary ECs of old mouse hippocampus tissue (Brain Endothelial Cells Are Exquisite Sensors of Age- Related Circulatory Cues, 2020). 
* ACER2 promotes tumour angiogenesis (expansion of capillary network to supply tumour with nutrients and oxygen) (https://doi.org/10.1096/fasebj.25.1_supplement.739.10)

**FGD6**

* Expressed in HUVEC but function is unknown
* Possibly involved with angiogenesis ((Arterioscler Thromb Vasc Biol. 2012;32:988-996.)

**TACC1**

*Also upregualted in aged ECs in mouse hippocampus tissue 

**PLAT**

* Involved in extracellular matrix processing (doi: 10.1152/ajpcell.00369.2011)

**TNRC6B**

* Reported to be important in microvascular maturation in developing lung (doi: 10.1016/j.ydbio.2017.07.018)


Here I plotted Violin plot of top 10 enhanced genes in cells from Old individuals. 

```{r,fig.width= 15, fig.height= 13, message=FALSE, warning=FALSE}
Idents(ec) <- "AgeGroup"

# enhanced genes in Old individuals
old.markers <- FindMarkers(ec,
                           ident.1 = "Old",
                           ident.2 = "Young",
                           only.pos = TRUE, 
                           min.pct = 0.25, 
                           logfc.threshold = 0.25,
                           test.use = "MAST")

old.markers_sig <- subset(old.markers, old.markers$p_val_adj < 0.05)

#old.markers_sig <- subset(old.markers,
#                          old.markers$pct.1 > 0.25 &
#                          old.markers$pct.2 < 0.25 )

old.markers_sig

VlnPlot(ec, features = rownames(old.markers_sig), group.by = "AgeGroup")
```

# Young-enhanced genes 

Three genes identified that satisfy adjusted p-value < 0.05. Requires further validation using different dataset with large number of ECs. Again, the log2FC is small, and none satisfy the pct1 and pct2 thresholds. 

**ABCG2**

* was enahnced in aged cells in mouse hippocampus
* Experiments have demonstrated that inhibition of ABCG2 impaired the survival of microvascular ECs under oxidative stress (PMID: 20829509)

**NFKBIA**

* NF-kappa-B inhibitor alpha
* Ageing is associated with greated NF-kB, reduced inhibitor of NF-kB (PMID: 18782346)

**ZNF385B**

* Zinc finger 


```{r,fig.width= 15, fig.height= 7, message=FALSE, warning=FALSE}
Idents(ec) <- "AgeGroup"

# enhanced genes in Young individuals
young.markers <- FindMarkers(ec,
                           ident.1 = "Young",
                           ident.2 = "Old",
                           only.pos = TRUE, 
                           min.pct = 0.25, 
                           logfc.threshold = 0.25,
                           test.use = "MAST")

young.markers_sig <- subset(young.markers,
                            young.markers$p_val_adj < 0.05)
young.markers
VlnPlot(ec, features = rownames(young.markers), group.by = "AgeGroup")
```

# Gene Set enrichment analysis (GSEA) with GO terms 

## Aged ECs

There are only 8 DEGs that satisfy the adjusted p-value < 0.05. Beucase there are so few genes, none of the pathways are significant. NES is enrichment score normalised to mean enrichment of random samples of the same size. Enrichment score reflects the degree ot which the genes in a gene set are overrepresented at the top or bottom of their entire ranked list of genes. ALthough none of the pathways are significant, the enriched pathways are interesting and point towards the right direction- complement: genes encoding components of complement system which is part of innate immune system. 

Also tried with longer list. To achieve longer list I obtained enhanced genes in aged ECs that satisfy non-adjusted p-value < 0.05. Of 193 DEGs, 171 were identified to be significant (unadjusted p-val < 0.05). Of these after formatting the dataset suited fro enrichment analysis, there were 168 DEGs. 

```{r}
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
library(annotables)
library(enrichplot)

# Return true/false where gene symbols in grch38 matches that in list of DEGs
idx <- grch38$symbol %in% rownames(old.markers)
# Subset for rows in grch38 that matches DEGs 
ids <- grch38[idx,]

# The gene names can map to more than one Ensembl ID (some genes change ID over time), 
# so we need to remove duplicate IDs prior to assessing enriched GO terms
non_duplicates <- which(duplicated(ids$symbol) == FALSE)
ids <- ids[non_duplicates, ] 

# Add gene column in DEG list 
old.markers["gene"] <- rownames(old.markers)

# Merge DEGs with subset of table from grch38
# res_ids is a data.frame containing DEGs and their Entrez number 
res_ids <- inner_join(old.markers, ids, by=c("gene" ="symbol")) 

# Create dataset containing significant genes 
sig_ids <- subset(res_ids, p_val < 0.05)
# remove NAs
res_entrez <- subset(sig_ids, entrez != "NA")
# Remove duplicates 
res_entrez <- res_entrez[which(duplicated(res_entrez$entrez) == F), ]
# Extract name and fold-changes
foldchanges <- res_entrez$avg_log2FC
# Name each fold change with the corresponding Entrez ID
names(foldchanges) <- res_entrez$entrez
# Order in decreasing order
foldchanges <- sort(foldchanges, decreasing = TRUE)
gseaGO<- gseGO(geneList = foldchanges, 
              OrgDb = org.Hs.eg.db, 
              ont = 'BP', 
              pAdjustMethod = "BH",
              minGSSize = 1, 
              pvalueCutoff = 1,
              scoreType = "pos") 

gseaGO_results <- as.data.frame(gseaGO@result)
gseaGO_tidy <- subset(gseaGO_results, subset = gseaGO_results$p.adjust<0.5)

# Plot barplot 
ggplot(gseaGO_tidy, aes(Description, NES)) + geom_col(aes(fill=p.adjust<0.2)) + theme_minimal() + coord_flip()
```

## GSEA with KEGG pathways 

```{r}

# Perform GSEA using KEGG gene sets 
gseaKEGG<- gseKEGG(geneList = foldchanges, 
              organism = "hsa", 
              minGSSize = 1, 
              pvalueCutoff = 1,
              scoreType = "pos")

# Extract results 
gseaKEGG_results <- gseaKEGG@result
gseaKEGG_tidy <- subset(gseaKEGG_results, subset = gseaKEGG_results$p.adjust<0.5)

#Barplot
ggplot(gseaKEGG_tidy, aes(Description, NES)) + geom_col(aes(fill=p.adjust <0.2)) + theme_minimal() + coord_flip()
```

## GSEA with MSigDB hallmarks and ontology terms 

* H = hallmark gene sets 
* c5 = gene sets derived from the GO biological process ontology

```{r}
# Load GMT file downloaded from MSigDB website 
h <- read.gmt("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/MSigDB_H")
c5 <- read.gmt("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/MSigDB_C5")

# Hallmarks
msig_h <- GSEA(foldchanges, 
               TERM2GENE=h,
               scoreType = "pos",
               minGSSize = 1,
               pvalueCutoff = 1,
               pAdjustMethod = "BH"
               )

msig_H_result <- data.frame(msig_h)
msig_H_tidy <- subset(msig_H_result, subset = msig_H_result$pvalue<0.5)

#Barplot
ggplot(msig_H_tidy, aes(Description, NES)) + geom_col(aes(fill=p.adjust <0.2)) + theme_minimal() + coord_flip()

# Gene ontology
msig_c5 <- GSEA(foldchanges, 
               TERM2GENE=c5,
               scoreType = "pos",
               minGSSize = 1,
               pvalueCutoff = 1,
               pAdjustMethod = "BH"
               )

msig_C5_result <- data.frame(msig_c5)
msig_C5_tidy <- subset(msig_C5_result, subset = msig_C5_result$p.adjust<0.6)

#Barplot
ggplot(msig_C5_tidy, aes(Description, NES)) + geom_col(aes(fill=p.adjust <0.2)) + theme_minimal() + coord_flip()

```
 
# ssGSEA

In ssGSEA, enrichment scores are calcualted across all individual cells for each pathway. Each ssGSEA enrichment score represents the degree to which the genes in a particular gene set are coordinately up- or down-regulated within a sample.There are 50 GSEA pathways. Within those, I looked at pathways often associated with ageing.  

```{r, fig.width=13, fig.height=10}
GS.hallmark <- getGeneSets(library = "H")
pathways <- enrichIt(ec, gene.sets = GS.hallmark, groups = 1000, cores = 2)
ec <- AddMetaData(ec, pathways)
VlnPlot(ec, features= c("HALLMARK_ANGIOGENESIS",
                        "HALLMARK_APOPTOSIS",
                        "HALLMARK_COMPLEMENT",
                        "HALLMARK_DNA_REPAIR",
                        "HALLMARK_HYPOXIA",
                        "HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",
                        "HALLMARK_INFLAMMATORY_RESPONSE",
                        "HALLMARK_COMPLEMENT"),
                        group.by = "AgeGroup")

DotPlot(ec, features= c("HALLMARK_ANGIOGENESIS",
                        "HALLMARK_APOPTOSIS",
                        "HALLMARK_COMPLEMENT",
                        "HALLMARK_DNA_REPAIR",
                        "HALLMARK_HYPOXIA",
                        "HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY",
                        "HALLMARK_INFLAMMATORY_RESPONSE"),
                        group.by = "AgeGroup")
```


## Higher expression in aged cells?

The study on ageing of mouse hippocampus showed that aged cells expressed more genes. Here, I tried checking this. It has been reported that transcriptional heterogeneity increases with age due to a general defect in transcriptional regulation (PMID: 29492790). 

```{r}
VlnPlot(ec, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), group.by="AgeGroup",ncol=3 )
```


# Hallmarks of ageing 

Hallmarks of Brain Aging: Adaptive and Pathological Modification by Metabolic States (2018)

1. Mitochondrial dysfunction
2. Accummulation of oxidatively damaged proteins, nucleic acids, lipds
3. dysregulated energy metabolism
4. Compromosed DNA repair 
5. Inflammation
6. Cell senescence? 

## Reduced number of pericytes in old? 

In young 43% and old 41% are pericytes. So, slightly less but difficult to test this with small number of cells. Would like to test this in larger datasets. 

```{r, fig.width=7}
ecpc <- readRDS(here("/Users/serachoi/Documents/Edinburgh/MScProject/Project/data/processed/ecpc04.RDS")) 

# Proportion of different cells in different age group 
prop_samples <- prop.table(table(ecpc$AgeGroup,Idents(ecpc)), margin=1)
prop_samples <- as.data.frame(prop_samples)
colnames(prop_samples) <- c("AgeGroup", "CellType", "proportion")

prop_samples

# Plot a barplot 
ggplot(as.data.frame(prop_samples), aes(x=AgeGroup, y=proportion, fill= CellType)) + geom_bar(stat="identity")+ theme_classic() + scale_fill_manual(values=mycoloursP[1:20])
```

## Mitochondrial dysfunction 

Ageing of brain is often associated with mitochondrial dysfunction. It is marked by:

* altered mitochondrial respiration
* decreased energy production
* increased oxidative damage to mitochondrial proteins and DNA

## Increased oxidative stress?

Increased production of reactive oxygen species and/or reduced antioxidant defecens cause accummulation of dysfunctional and aggregated proteins in cells. Hypoxia? 

## Angiogenesis 

Association of angeiogensis and age has been studied in heart. Angiogenesis is impaired in aged cells (microvessel ECs of heart). Due to this decline in microvascular density, there is decreased blood flow, impaired response to hypoxia, worsened ischemic tissue injury. During cardiac muscle growth, an imbalance between growth of muscle and impaired angiogenesis is likely to contribute towards cardiac failure. Ageing leads to dysregulated expression of both positive and negative regulators of angiogenesis in the heart. 


## Figure 

```{r, fig.height=2.5, fig.width=11}
features <- c(rownames(old.markers_sig),rownames(young.markers))

pdf("/Users/serachoi/Documents/Edinburgh/MScProject/Project/outs/age/Fig6_DEG.pdf", height = 2.5, width = 11)
VlnPlot(ec, features = features, stack = TRUE, sort = TRUE, flip = F, cols = mycoloursP[15:30]) + theme(legend.position = "none", strip.text.x.top = element_text(angle = 0))
dev.off()
```





